api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -583,6 +583,30 @@ message HttpConnectionManager {
   // *not* the deprecated but similarly named :ref:`stream_error_on_invalid_http_messaging
   // <envoy_v3_api_field_config.core.v3.Http2ProtocolOptions.stream_error_on_invalid_http_messaging>`
   google.protobuf.BoolValue stream_error_on_invalid_http_message = 40;
+
+  // The configuration of the original IP detection extension.
+  //
+  // If not set, Envoy uses the default remote IP detection mechanism.
```

Comment:
Sure, waiting for @markdroth's feedback.

Reply:
As per my comment elsewhere, I think we should move the existing XFF functionality into an extension, and just make that the default.  Then the comments here can say that this extension dictates how the xDS client should find the original IP address, and it can list the supported extensions with links to the docs for each of those extensions.

Also, would it be useful to allow inhibiting *all* functionality here?  If so, we may want to allow setting the `TypedExtensionConfig` but leaving it empty as a way to inhibit even the normal XFF functionality.

Reply:
There is a bool config option to disable the current XFF behavior, see https://github.com/envoyproxy/envoy/blob/def62d34ad344732cd3ed08f121c4a577b7050cf/api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#L427

It would be good to nudge the API defaults so they are a bit more consistent.

Reply:
Can you ref link also to the xff extension itself for clarity? cc @phlax for a new extension type that will need to get plumbed into the docs.